### PR TITLE
fix disabled shadow inconsistency

### PR DESCRIPTION
**Pull request by star-e**
Fix missing csm info caused by mismatching set/get pair.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp [hunk 1]
```diff
@@ -873,6 +850,50 @@ const BuiltinCascadedShadowMap *getBuiltinShadowCSM(
...
+
+    { // find or create csm info
+        const BuiltinCascadedShadowMapKey key{&camera, mainLight};
+        auto iter = ppl.builtinCSMs.find(key);
+        if (iter != ppl.builtinCSMs.end()) {
+            result = &iter->second;
+        } else {
+            result = addBuiltinCSMInfo(ppl, pplSceneData, camera, mainLight, key, csmLayers);
+        }
+    }
```
Comment (star-e) [inline native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp]: Always insert csm info.

File: native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp [hunk 2]
```diff
@@ -873,6 +850,50 @@ const BuiltinCascadedShadowMap *getBuiltinShadowCSM(
...
+    CC_ENSURES(result);
+
+    // shadow not enabled
+    if (!shadows.isEnabled()) {
+        return nullptr;
+    }
+    // shadow type is planar
+    if (shadows.getType() == scene::ShadowType::PLANAR) {
+        return nullptr;
+    }
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp]: I think it is better to check them as soon as possible.

Comment (star-e) [reply native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp]: fixed.

File: native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp [hunk 3]
```diff
@@ -873,6 +849,51 @@ const BuiltinCascadedShadowMap *getBuiltinShadowCSM(
...
+        } else {
+            result = addBuiltinCSMInfo(ppl, pplSceneData, camera, mainLight, key, csmLayers);
+        }
+    }
+    CC_ENSURES(result);
+
+    // shadow not enabled
+    if (!shadows.isEnabled()) {
+        return nullptr;
+    }
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp]: Can also check it as soon as possible.

Comment (star-e) [reply native/cocos/renderer/pipeline/custom/NativeBuiltinUtils.cpp]: Check shadow not enabled earlier caused the mismatch. When doing the rendering, csm info is always needed.

**Review by minggo — APPROVED** [review]

**Pull request merged**